pan3d/dataset_builder.py

```diff
@@ -124,6 +141,18 @@ def data_array_name(self) -> Optional[str]:
 
     @data_array_name.setter
     def data_array_name(self, data_array_name: Optional[str]) -> None:
+        if data_array_name is not None:
+            if not isinstance(data_array_name, str):
+                raise TypeError("Type of data_array_name must be str or None.")
+            if self.dataset is None:
+                raise ValueError(
+                    "Cannot set data array name without setting dataset info first."
+                )
+            acceptable_values = list(self.dataset.data_vars.keys())
+            if data_array_name not in acceptable_values:
```

Comment:
Why not just `if data_array_name not in self.dataset.data_vars:`? And then generate the keys in case of error...